org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaThread.scala

```diff
@@ -47,10 +48,12 @@ abstract class ScalaThread private(target: ScalaDebugTarget, val threadRef: Thre
 
   // Members declared in org.eclipse.debug.core.model.IStep
 
-  override def canStepInto: Boolean = suspended // TODO: need real logic
-  override def canStepOver: Boolean = suspended // TODO: need real logic
-  override def canStepReturn: Boolean = suspended // TODO: need real logic
+  override def canStepInto: Boolean = canStep
+  override def canStepOver: Boolean = canStep
+  override def canStepReturn: Boolean = canStep
   override def isStepping: Boolean = ???
+  private def canStep = suspended && // TODO: needs real logic
```

Comment:
I don't know what is the real logic which an author had in mind. I only added one condition. Thus I didn't remove TODOs.